doc/guides/modules.rst

```diff
@@ -10,14 +10,380 @@ system those are called *modules*. These modules must be integrated with the
+     the external project is not hosted in an upstream Git repository.
+
+The commit message is expected to identify the upstream repository, the
+version to which the module is updated (upstream version, tag, commit,
+etc.), and the reason for the doing the update.
+
+**Updating modules by merging the upstream main branch:**
+Upstream changes brought in by performing a Git merge of the upstream main
+branch submitting the result in pull request against the module main branch.
+This approach is applicable to modules with an upstream project Git repository.
+The main advantages of this approach is that the upstream repository history
+(that is, the original commit SHAs) is preserved in the module repository. The
+downside of this approach is that an additional merge commit is generated in
+the downstream main branch.
+
+
+Contributing to Zephyr modules
+******************************
+
+.. _modules_contributing:
+
+
+Individual Roles & Responsibilities
+===================================
+
+To facilitate management of Zephyr module repositories, the following
+individual roles are defined.
+
+**Administrator:** Each Zephyr module shall have an administrator
+who is responsible for managing access to the module repository,
+for example, for adding individuals as Collaborators in the repository
+at the request of the module owner. Module administrators are
+members of the Administrators team, that is a group of project
+members with admin rights to Module Github repositories.
+
+**Module owner:** Each module shall have a module code owner. Module
+owners will have the overall responsibility of the contents of a
+Zephyr module repository. In particular, a module owner will:
+
+* coordinate code reviewing in the module repository
+* be the default assignee in pull-requests against the repository's
+  main branch
+* request additional collaborators to be added to the repository, as
+  they see fit
+* regularly synchronize the module repository with its upstream
+  counterpart following the policies described in
+  :ref:`modules_synchronization`
+* be aware of security vulnerability issues in the external project
+  and update the module repository to include security fixes, as
+  soon as the fixes are available in the upstream code base
+* list any known security vulnerability issues, present in the
+  module code base, in Zephyr release notes.
+
+
+  .. note::
+
+     Module owners are not required to be Zephyr
+     :ref:`Maintainers <project_roles>`.
+
+**Merger:** The Zephyr Release Engineering team has the right and the
+responsibility to merge approved pull requests in the main branch of a
+module repository.
+
+  .. note::
+
+     The merging of pull requests in the main branch of a module
+     repository must be coupled with the corresponding manifest
+     file update in the Zephyr main tree.
+
+
+Maintaining module code base
+============================
+
+Updates in the Zephyr main tree, for example, in public Zephyr APIs,
```

Comment:
This reminds me, we haven't captured the fact that we'd like glue code to live in the main zephyr tree in this PR, will this be done in a separate PR?

Reply:
Yes!

Reply:
Made a note for this. Will be done in future amending of this doc.